Client/Frontend/Autofill/CreditCard/ViewComponents/CreditCardInputField.swift

```diff
@@ -0,0 +1,195 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+import SwiftUI
+import Shared
+
+enum CreditCardInputType {
+    case name, number, expiration
+}
+
+struct CreditCardInputField: View {
```

Comment:
There is a lot of logic that could be tested in this View I think needs a viewModel to move all the logic and add unit test easily

Reply:
Can we instead keep them inside CreditCardInputField and test it? The two funcs I see that can be moved are `separate` and `countNumbersIn`, but both also directly deal with the view and I think it might be okay keeping it there.